mods/beds/functions.lua

```diff
@@ -81,6 +81,21 @@ local function lay_down(player, pos, bed_pos, state, skip)
 
 	-- lay down
 	else
+
+		-- Check if bed is occupied
+		for _, other_pos in pairs(beds.bed_position) do
+			if vector.distance(bed_pos, other_pos) < 0.1 then
+				minetest.chat_send_player(name, S("This bed is already occupied."))
+				return false
+			end
+		end
+
+		-- Check if player is moving
+		if vector.length(player:get_player_velocity()) > 0.001 then
```

Comment:
I use the same solution in my game.  > 0 is enough.  > 0.001 is useless.

Reply:
Done

Reply:
Not yet :)